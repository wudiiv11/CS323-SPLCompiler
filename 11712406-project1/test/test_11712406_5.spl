struct a {
    int a = 0x123FfA321;
    char a = '\x0F';
    float a;
} a;

int func(int val) {
    val = val + t();
    return (val / '2');
}

int main() 
{
    int a, b;
    a = func(slist[func(0)].xx_id - 1) + 52 / a;
    if (a) {  }
    {
        struct s2 {
            int _val = 2;
        } c;
    }
    return c;
}