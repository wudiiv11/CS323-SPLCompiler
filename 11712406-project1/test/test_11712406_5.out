Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        StructSpecifier (1)
          STRUCT
          ID: a
          LC
          DefList (2)
            Def (2)
              Specifier (2)
                TYPE: int
              DecList (2)
                Dec (2)
                  VarDec (2)
                    ID: a
                  ASSIGN
                  Exp (2)
                    INT: 603956001
              SEMI
            DefList (3)
              Def (3)
                Specifier (3)
                  TYPE: char
                DecList (3)
                  Dec (3)
                    VarDec (3)
                      ID: a
                    ASSIGN
                    Exp (3)
                      CHAR: \x0F
                SEMI
              DefList (4)
                Def (4)
                  Specifier (4)
                    TYPE: float
                  DecList (4)
                    Dec (4)
                      VarDec (4)
                        ID: a
                  SEMI
          RC
      ExtDecList (5)
        VarDec (5)
          ID: a
      SEMI
    ExtDefList (7)
      ExtDef (7)
        Specifier (7)
          TYPE: int
        FunDec (7)
          ID: func
          LP
          VarList (7)
            ParamDec (7)
              Specifier (7)
                TYPE: int
              VarDec (7)
                ID: val
          RP
        CompSt (7)
          LC
          StmtList (8)
            Stmt (8)
              Exp (8)
                Exp (8)
                  ID: val
                ASSIGN
                Exp (8)
                  Exp (8)
                    ID: val
                  PLUS
                  Exp (8)
                    ID: t
                    LP
                    RP
              SEMI
            StmtList (9)
              Stmt (9)
                RETURN
                Exp (9)
                  LP
                  Exp (9)
                    Exp (9)
                      ID: val
                    DIV
                    Exp (9)
                      CHAR: 2
                  RP
                SEMI
          RC
      ExtDefList (12)
        ExtDef (12)
          Specifier (12)
            TYPE: int
          FunDec (12)
            ID: main
            LP
            RP
          CompSt (13)
            LC
            DefList (14)
              Def (14)
                Specifier (14)
                  TYPE: int
                DecList (14)
                  Dec (14)
                    VarDec (14)
                      ID: a
                  COMMA
                  DecList (14)
                    Dec (14)
                      VarDec (14)
                        ID: b
                SEMI
            StmtList (15)
              Stmt (15)
                Exp (15)
                  Exp (15)
                    ID: a
                  ASSIGN
                  Exp (15)
                    Exp (15)
                      ID: func
                      LP
                      Args (15)
                        Exp (15)
                          Exp (15)
                            Exp (15)
                              Exp (15)
                                ID: slist
                              LB
                              Exp (15)
                                ID: func
                                LP
                                Args (15)
                                  Exp (15)
                                    INT: 0
                                RP
                              RB
                            DOT
                            ID: xx_id
                          MINUS
                          Exp (15)
                            INT: 1
                      RP
                    PLUS
                    Exp (15)
                      Exp (15)
                        INT: 52
                      DIV
                      Exp (15)
                        ID: a
                SEMI
              StmtList (16)
                Stmt (16)
                  IF
                  LP
                  Exp (16)
                    ID: a
                  RP
                  Stmt (16)
                    CompSt (16)
                      LC
                      RC
                StmtList (17)
                  Stmt (17)
                    CompSt (17)
                      LC
                      DefList (18)
                        Def (18)
                          Specifier (18)
                            StructSpecifier (18)
                              STRUCT
                              ID: s2
                              LC
                              DefList (19)
                                Def (19)
                                  Specifier (19)
                                    TYPE: int
                                  DecList (19)
                                    Dec (19)
                                      VarDec (19)
                                        ID: _val
                                      ASSIGN
                                      Exp (19)
                                        INT: 2
                                  SEMI
                              RC
                          DecList (20)
                            Dec (20)
                              VarDec (20)
                                ID: c
                          SEMI
                      RC
                  StmtList (22)
                    Stmt (22)
                      RETURN
                      Exp (22)
                        ID: c
                      SEMI
            RC
